servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -968,6 +970,14 @@ private static <T, A> Publisher<? extends Collection<ServiceDiscovererEvent<T>>>
         });
     }
 
+    private static boolean shouldRevokeState(final Throwable t) {
+        // ISE => Subscriber exceptions (downstream of retry)
+        return t instanceof SrvAddressRemovedException || t instanceof IllegalStateException ||
+                t instanceof ClosedDnsServiceDiscovererException || (NX_DOMAIN_INVALIDATES && t.getCause() != null &&
+                t.getCause() instanceof DnsErrorCauseException &&
+                NXDOMAIN.equals(((DnsErrorCauseException) t.getCause()).getCode()));
```

Comment:
```suggestion
        return t instanceof SrvAddressRemovedException || t instanceof IllegalStateException ||
                t instanceof ClosedDnsServiceDiscovererException || (NX_DOMAIN_INVALIDATES &&
                t.getCause() instanceof DnsErrorCauseException &&
                NXDOMAIN.equals(((DnsErrorCauseException) t.getCause()).getCode()));
```
The `instanceof` check will ensure the value is not null.